### DISCUSSION TITLE
How to disable AdditionalProperties being generated in Swagger

### DISCUSSION THREAD

**Discussion by nitinpasula** [swagger-api/swagger-ui#7975] [Q&A]
Hi, We are using RestEasy to build our rest services and swagger annotations for documentation. We are upgrading from a very old version of swagger UI 2.x to 4.10.3. Seeing below in few object responses, which were not showing before, Is there any way to disable them being generated. 

{
  "additionalProp1": {},
  "additionalProp2": {},
  "additionalProp3": {}
}

**Comment by Lukas-Pacl** [comment]
I had the same problem with using the .NET 6 Swashbuckle package and solved it by creating a custom schema filter that disabled additional properties. From research, I found out that since some version of swagger AdditionalProperties is set globally to True instead of False for some reason.

Here is my schema filter implementation
```
public class AdditionalPropertiesSchemaFilter : ISchemaFilter
{
    public void Apply(OpenApiSchema schema, SchemaFilterContext context)
    {
        schema.AdditionalPropertiesAllowed = false;
    }
}
```
and here how to use it with swagger in .NET 6

```
services.AddSwaggerGen(c =>
{
    ...
    c.SchemaFilter<AdditionalPropertiesSchemaFilter>();
});
```
I hope this will at least help point you in the right direction

**Comment by kjkrum** [comment]
Note that `OpenApiSchema` has two related properties: `AdditionalPropertiesAllowed` and `AdditionalProperties`. If you just don't want to see `"additionalProperty1"` etc. in the Swagger UI, leave `AdditionalPropertiesAllowed` true and set `AdditionalProperties` to null. I think `AdditionalPropertiesAllowed` allows for future changes, while `AdditionalProperties` tells the consumer to _expect_ additional properties, for example with types like `ProblemDetails`.
